lib/util/config_utils.js

```diff
@@ -28,13 +28,19 @@ shaka.util.ConfigUtils = class {
   static mergeConfigObjects(destination, source, template, overrides, path) {
     goog.asserts.assert(destination, 'Destination config must not be null!');
 
+    // If true, override the template.
+    const overrideTemplate = path in overrides;
+
     // If true, treat the source as a generic object to be copied without
     // descending more deeply.
-    const genericObject = template.constructor == Object &&
+    let genericObject = false;
+    if (overrideTemplate) {
+      genericObject = template.constructor == Object &&
+        Object.keys(overrides).length == 0;
```

Comment:
This is not right.  `overrides` always has keys (eg ".drm.advanced"), and I don't see why you would check it here.

Reply:
It’s the only solution that I found to fix the tests